src/dmd/root/file.d

```diff
@@ -309,10 +299,20 @@ nothrow:
 
     /* Set buffer
      */
-    extern (C++) void setbuffer(void* buffer, size_t len)
+    extern (C++) void setbuffer(void* buffer, size_t len, bool transferOwnership = false)
     {
         this.buffer = cast(ubyte*)buffer;
         this.len = len;
+        this._ref = !transferOwnership;
+    }
+
+    /* Set buffer (incl. ownership) and reset the specified OutBuffer.
+     */
+    extern (C++) void hijackBufferFrom(ref OutBuffer buffer)
```

Comment:
The word "hijack" has negative connotations, besides, I don't know what it means in this context. Perhaps "transfer" instead?